src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
```

Comment:
you can use logical operand
```suggestion
    let result = initialValue || this[0];
```

Reply:
I thought to do the same, but some tests fail because of this(

Reply:
What test?